app/src/main/java/org/kiwix/kiwixmobile/ZimContentProvider.java

```diff
@@ -387,25 +388,12 @@ private ParcelFileDescriptor loadContent(Uri uri) throws FileNotFoundException {
     return (pipe[0]);
   }
 
-  private ParcelFileDescriptor saveVideoToCache(Uri uri) throws IOException {
-    String filePath = getFilePath(uri);
-
-    String fileName = uri.toString();
-    fileName = fileName.substring(fileName.lastIndexOf('/') + 1, fileName.length());
-
-    File f = new File(FileUtils.getFileCacheDir(getContext()), fileName);
-
-    JNIKiwixString mime = new JNIKiwixString();
-    JNIKiwixString title = new JNIKiwixString();
-    JNIKiwixInt size = new JNIKiwixInt();
-    byte[] data = currentJNIReader.getContent(filePath, title, mime, size);
-
-    FileOutputStream out = new FileOutputStream(f);
-
-    out.write(data, 0, data.length);
-    out.flush();
-
-    return ParcelFileDescriptor.open(f, ParcelFileDescriptor.MODE_READ_ONLY);
+  // Return a file descriptor of the video within the ZIM file
+  private ParcelFileDescriptor loadVideo(Uri uri) throws IOException {
+    Pair pair = currentJNIReader.getDirectAccessInformation(getFilePath(uri));
```

Comment:
Be careful. If the a direct access is not available (because the zim file is split in the middle of the content), `pair.filename` will be an empty string and `pair.offset` will be 0.

You should test the value of `pair.filename` before trying to opening it.
You have to handle somehow the case where `pair.filename` is empty ("return" 404, copy the video in a temp file and play from it , ...)